homeassistant/components/recorder/purge.py

```diff
@@ -69,8 +69,7 @@ def purge_old_data(
             return False
 
         _LOGGER.warning("Error purging history: %s", err)
-    except SQLAlchemyError as err:
```

Comment:
We catch this in `__init__.py` since `SQLAlchemyError` is too broad here and we want to know the difference of what is a database error and what is not.